Resolve sonar issues on master

Neeharika-Sompalli opened the pull request:
Closes #2687 

Resolves most of the sonar issues on `master`

Issues on master reduced to 71 from 290 https://sonarcloud.io/project/issues?branch=branch-resolve-sonar-issues&id=com.hedera.hashgraph%3Ahedera-services&resolved=false&types=CODE_SMELL

tinker-michaelj commented on hedera-node/src/main/java/com/hedera/services/files/store/BytesStoreAdapter.java:
```diff
@@ -64,6 +65,23 @@ public int size() {
 		return delegate.size();
 	}
 
+	@Override
```
Why did we add `equals()` and `hashCode()` here? This class won't ever be used in a collection anywhere, right?

Neeharika-Sompalli replied:
There was a sonar code smell stating , `Extend a class that overrides equals and add fields without overriding equals in the subclass, you run the risk of non-equivalent instances of your subclass being seen as equal, because only the superclass fields will be considered in the equality test`. I believe that is because this class is extended from `AbstractMap`.  What do you think ?

Neeharika-Sompalli replied:
removed them. Ty!

tinker-michaelj commented on hedera-node/src/main/java/com/hedera/services/files/store/FcBlobsBytesStore.java:
```diff
@@ -132,4 +133,19 @@ public int size() {
 				.collect(toSet());
 	}
+
+	@Override
```
Same question here. 😁

Neeharika-Sompalli replied:
same issue was here 🙂

Neeharika-Sompalli replied:
removed them. Ty!

tinker-michaelj commented on hedera-node/src/main/java/com/hedera/services/ledger/accounts/BackingNfts.java:
```diff
@@ -60,9 +60,7 @@ public MerkleUniqueToken getImmutableRef(NftId id) {
-		if (!currentNfts.containsKey(key)) {
-			currentNfts.put(key, nft);
-		}
+		currentNfts.computeIfAbsent(key, k -> nft);
```
I would slightly prefer to use the more verbose `containsKey` to avoid creating the lambda in the second argument. ..not sure if it matters though.

Neeharika-Sompalli replied:
Modified to be using as before

tinker-michaelj reviewed (approved):
LGTM, aside from the new `equals()`/`hashCode()` methods in classes that don't seem to have any sense of logical identity other than object identity.

anighanta commented on hedera-node/src/main/java/com/hedera/services/context/primitives/StateView.java:
```diff
@@ -502,6 +502,7 @@ public long numNftsOwnedBy(AccountID target) {
 			var adminKey = JKey.mapJKey(contract.getAccountKey());
 			info.setAdminKey(adminKey);
 		} catch (Exception ignore) {
+			return Optional.of(info.build());
```
nit : may be we can log a message instead of a return here.

Neeharika-Sompalli replied:
Not sure if it will emit lot of logs, if a log is added(as the exception is ignored previously)

anighanta commented on hedera-node/src/main/java/com/hedera/services/ledger/HederaLedger.java:
```diff
@@ -347,7 +347,7 @@ public ResponseCodeEnum doTokenTransfer(
 	) {
-		var validity = OK;
+		ResponseCodeEnum validity;
 		validity = adjustTokenBalance(from, tId, -adjustment);
```
nit : we can combine line 350 and 351

Neeharika-Sompalli replied:
Done

qnswirlds commented on hedera-node/src/main/java/com/hedera/services/txns/consensus/TopicUpdateTransitionLogic.java:
```diff
@@ -63,7 +63,7 @@
 	private static final Logger log = LogManager.getLogger(TopicUpdateTransitionLogic.class);
 
-	private final Function<TransactionBody, ResponseCodeEnum> PRE_SIGNATURE_VALIDATION_SEMANTIC_CHECK =
+	private final Function<TransactionBody, ResponseCodeEnum> preSignatureValidationSemanticCheck =
 			this::validatePreSignatureValidation;
```
Remove this, return directly `this::validatePreSignatureValidation` in `semanticCheck`

qnswirlds commented on hedera-node/src/main/java/com/hedera/services/txns/contract/ContractCallTransitionLogic.java:
```diff
@@ -55,7 +55,7 @@
 	private final SigImpactHistorian sigImpactHistorian;
 
-	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validateSemantics;
+	private final Function<TransactionBody, ResponseCodeEnum> semanticCheck = this::validateSemantics;
```
Remove this, return directly this::validateSemantics in semanticCheck

qnswirlds commented on hedera-node/src/main/java/com/hedera/services/txns/contract/ContractCreateTransitionLogic.java:
```diff
@@ -70,7 +70,7 @@
 	private final SigImpactHistorian sigImpactHistorian;
 
-	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
+	private final Function<TransactionBody, ResponseCodeEnum> semanticCheck = this::validate;
```
Remove this, return directly this::validate in semanticCheck

qnswirlds commented on hedera-node/src/main/java/com/hedera/services/txns/contract/ContractDeleteTransitionLogic.java:
```diff
@@ -56,7 +56,7 @@
 	private final Supplier<MerkleMap<EntityNum, MerkleAccount>> contracts;
 
-	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
+	private final Function<TransactionBody, ResponseCodeEnum> semanticCheck = this::validate;
```
Remove this, return directly this::validate in semanticCheck

qnswirlds commented on hedera-node/src/main/java/com/hedera/services/txns/contract/ContractSysDelTransitionLogic.java:
```diff
@@ -55,7 +55,7 @@
 	private final Supplier<MerkleMap<EntityNum, MerkleAccount>> contracts;
 
-	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
+	private final Function<TransactionBody, ResponseCodeEnum> semanticCheck = this::validate;
```
Remove this, return directly this::validate in semanticCheck

qnswirlds commented on hedera-node/src/main/java/com/hedera/services/txns/contract/ContractSysUndelTransitionLogic.java:
```diff
@@ -55,7 +55,7 @@
 	private final Supplier<MerkleMap<EntityNum, MerkleAccount>> contracts;
 
-	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
+	private final Function<TransactionBody, ResponseCodeEnum> semanticCheck = this::validate;
```
Remove this, return directly this::validate in semanticCheck

qnswirlds commented on hedera-node/src/main/java/com/hedera/services/txns/contract/ContractUpdateTransitionLogic.java:
```diff
@@ -56,7 +56,7 @@
 	private final Supplier<MerkleMap<EntityNum, MerkleAccount>> contracts;
 
-	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
+	private final Function<TransactionBody, ResponseCodeEnum> semanticCheck = this::validate;
```
Remove this, return directly this::validate in semanticCheck

qnswirlds commented on hedera-node/src/main/java/com/hedera/services/txns/crypto/CryptoCreateTransitionLogic.java:
```diff
@@ -67,7 +67,7 @@
 	private static final Logger log = LogManager.getLogger(CryptoCreateTransitionLogic.class);
 
-	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
+	private final Function<TransactionBody, ResponseCodeEnum> semanticCheck = this::validate;
```
Remove this, return directly this::validate in semanticCheck

qnswirlds commented on hedera-node/src/main/java/com/hedera/services/txns/crypto/CryptoDeleteTransitionLogic.java:
```diff
@@ -60,7 +60,7 @@
 	private static final Logger log = LogManager.getLogger(CryptoDeleteTransitionLogic.class);
 
-	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
+	private final Function<TransactionBody, ResponseCodeEnum> semanticCheck = this::validate;
```
Remove this, return directly this::validate in semanticCheck

qnswirlds commented on hedera-node/src/main/java/com/hedera/services/txns/crypto/CryptoUpdateTransitionLogic.java:
```diff
@@ -73,7 +73,7 @@
 	private static final EnumSet<AccountProperty> EXPIRY_ONLY = EnumSet.of(AccountProperty.EXPIRY);
 
-	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
+	private final Function<TransactionBody, ResponseCodeEnum> semanticCheck = this::validate;
```
Remove this, return directly this::validate in semanticCheck

qnswirlds commented on hedera-node/src/main/java/com/hedera/services/txns/file/FileCreateTransitionLogic.java:
```diff
@@ -59,7 +59,7 @@
 	private final TransactionContext txnCtx;
 
-	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
+	private final Function<TransactionBody, ResponseCodeEnum> semanticCheck = this::validate;
```
Remove this, return directly this::validate in semanticCheck

qnswirlds commented on hedera-node/src/main/java/com/hedera/services/txns/file/FileUpdateTransitionLogic.java:
```diff
@@ -65,7 +65,7 @@
 	private static final Logger log = LogManager.getLogger(FileUpdateTransitionLogic.class);
 
-	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
+	private final Function<TransactionBody, ResponseCodeEnum> semanticCheck = this::validate;
```
Remove this, return directly this::validate in semanticCheck

qnswirlds commented on hedera-node/src/main/java/com/hedera/services/txns/schedule/ScheduleSignTransitionLogic.java:
```diff
@@ -49,7 +49,7 @@
 	private static final Logger log = LogManager.getLogger(ScheduleSignTransitionLogic.class);
 
-	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
+	private final Function<TransactionBody, ResponseCodeEnum> semanticCheck = this::validate;
```
Remove this, return directly this::validate in semanticCheck

qnswirlds commented on hedera-node/src/main/java/com/hedera/services/txns/token/TokenAssociateTransitionLogic.java:
```diff
@@ -48,7 +48,7 @@
 @Singleton
 public class TokenAssociateTransitionLogic implements TransitionLogic {
-	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
+	private final Function<TransactionBody, ResponseCodeEnum> semanticCheck = this::validate;
```
Remove this, return directly this::validate in semanticCheck

qnswirlds commented on hedera-node/src/main/java/com/hedera/services/txns/token/TokenBurnTransitionLogic.java:
```diff
@@ -54,7 +54,7 @@
 	private final GlobalDynamicProperties dynamicProperties;
 
-	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
+	private final Function<TransactionBody, ResponseCodeEnum> semanticCheck = this::validate;
```
Remove this, return directly this::validate in semanticCheck

qnswirlds commented on hedera-node/src/main/java/com/hedera/services/txns/token/TokenCreateTransitionLogic.java:
```diff
@@ -59,7 +59,7 @@
 @Singleton
 public class TokenCreateTransitionLogic implements TransitionLogic {
-	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
+	private final Function<TransactionBody, ResponseCodeEnum> semanticCheck = this::validate;
```
Remove this, return directly this::validate in semanticCheck

qnswirlds commented on hedera-node/src/main/java/com/hedera/services/txns/token/TokenDeleteTransitionLogic.java:
```diff
@@ -42,7 +42,7 @@
 @Singleton
 public class TokenDeleteTransitionLogic implements TransitionLogic {
-	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
+	private final Function<TransactionBody, ResponseCodeEnum> semanticCheck = this::validate;
```
Remove this, return directly this::validate in semanticCheck

qnswirlds commented on hedera-node/src/main/java/com/hedera/services/txns/token/TokenDissociateTransitionLogic.java:
```diff
@@ -50,7 +50,7 @@
 @Singleton
 public class TokenDissociateTransitionLogic implements TransitionLogic {
-	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
+	private final Function<TransactionBody, ResponseCodeEnum> semanticCheck = this::validate;
```
Remove this, return directly this::validate in semanticCheck

qnswirlds commented on hedera-node/src/main/java/com/hedera/services/txns/token/TokenFeeScheduleUpdateTransitionLogic.java:
```diff
@@ -53,7 +53,7 @@
 	private final GlobalDynamicProperties dynamicProperties;
 
-	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
+	private final Function<TransactionBody, ResponseCodeEnum> semanticCheck = this::validate;
```
Remove this, return directly this::validate in semanticCheck

qnswirlds commented on hedera-node/src/main/java/com/hedera/services/txns/token/TokenFreezeTransitionLogic.java:
```diff
@@ -41,7 +41,7 @@
 @Singleton
 public class TokenFreezeTransitionLogic implements TransitionLogic {
-	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
+	private final Function<TransactionBody, ResponseCodeEnum> semanticCheck = this::validate;
```
Remove this, return directly this::validate in semanticCheck

qnswirlds commented on hedera-node/src/main/java/com/hedera/services/txns/token/TokenGrantKycTransitionLogic.java:
```diff
@@ -41,7 +41,7 @@
 @Singleton
 public class TokenGrantKycTransitionLogic implements TransitionLogic {
-	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
+	private final Function<TransactionBody, ResponseCodeEnum> semanticCheck = this::validate;
```
Remove this, return directly this::validate in semanticCheck

qnswirlds commented on hedera-node/src/main/java/com/hedera/services/txns/token/TokenMintTransitionLogic.java:
```diff
@@ -53,7 +53,7 @@
 @Singleton
 public class TokenMintTransitionLogic implements TransitionLogic {
-	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
+	private final Function<TransactionBody, ResponseCodeEnum> semanticCheck = this::validate;
```
Remove this, return directly this::validate in semanticCheck

qnswirlds commented on hedera-node/src/main/java/com/hedera/services/txns/token/TokenRevokeKycTransitionLogic.java:
```diff
@@ -41,7 +41,7 @@
 @Singleton
 public class TokenRevokeKycTransitionLogic implements TransitionLogic {
-	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
+	private final Function<TransactionBody, ResponseCodeEnum> semanticCheck = this::validate;
```
Remove this, return directly this::validate in semanticCheck

qnswirlds commented on hedera-node/src/main/java/com/hedera/services/txns/token/TokenUnfreezeTransitionLogic.java:
```diff
@@ -41,7 +41,7 @@
 @Singleton
 public class TokenUnfreezeTransitionLogic implements TransitionLogic {
-	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
+	private final Function<TransactionBody, ResponseCodeEnum> semanticCheck = this::validate;
```
Remove this, return directly this::validate in semanticCheck

qnswirlds commented on hedera-node/src/main/java/com/hedera/services/txns/token/TokenUpdateTransitionLogic.java:
```diff
@@ -66,7 +66,7 @@
 	private static final Logger log = LogManager.getLogger(TokenUpdateTransitionLogic.class);
 
-	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
+	private final Function<TransactionBody, ResponseCodeEnum> semanticCheck = this::validate;
```
Remove this, return directly this::validate in semanticCheck

qnswirlds commented on hedera-node/src/main/java/com/hedera/services/txns/token/TokenWipeTransitionLogic.java:
```diff
@@ -54,7 +54,7 @@
 	private final OptionValidator validator;
 	private final GlobalDynamicProperties dynamicProperties;
-	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
+	private final Function<TransactionBody, ResponseCodeEnum> semanticCheck = this::validate;
```
Remove this, return directly this::validate in semanticCheck

qnswirlds commented on hedera-node/src/test/java/com/hedera/services/files/EntityExpiryMapFactoryTest.java:
```diff
@@ -125,8 +125,9 @@ void toLongPropagatesNull() {
 	@Test
 	void throwsIaeOnNonsense() {
+		final var bytes = "wtf".getBytes();
 		// expect:
```
nit: remove also comment (replace by a blank line)

Neeharika-Sompalli replied:
Many tests have comments for readability. If these are decided to be removed, might be better to do everything in separate PR.

qnswirlds reviewed (commented):
Great work! Please see my inline comments. Please close [these issues](https://github.com/hashgraph/hedera-services/issues?q=is%3Aopen+is%3Aissue+author%3Aqnswirlds+label%3A%22Code+cleanup%22+label%3Asonar) where appropriate as well. Thanks!

Neeharika-Sompalli commented on hapi-utils/src/main/java/com/hedera/services/legacy/proto/utils/CommonUtils.java:
```diff
@@ -117,6 +121,7 @@ public static void writeToFile(String path, byte[] data, boolean append) throws
 			fos.write(data);
 			fos.flush();
 		} catch (IOException e) {
+			log.error("Exception caused while writing to file", e);
```
Removed the method.

Neeharika-Sompalli commented on hapi-utils/src/main/java/com/hedera/services/legacy/proto/utils/CommonUtils.java:
```diff
@@ -239,6 +244,7 @@ public static String toReadableString(Transaction grpcTransaction) throws Invali
 			rv = "body=" + TextFormat.shortDebugString(body) + "; sigs="
 					+ TextFormat.shortDebugString(extractSignatureMap(grpcTransaction));
 		} catch (InvalidProtocolBufferException e) {
+			log.error("Exception caused while reading grpcTransaction to string", e);
```
Removed the method on master

Neeharika-Sompalli commented:
> these issues

@qnswirlds  Thanks for your time to review. Addressed review comments and resolved them. Also closed many issues from [here](https://github.com/hashgraph/hedera-services/issues?q=is%3Aopen+is%3Aissue+author%3Aqnswirlds+label%3A%22Code+cleanup%22+label%3Asonar) as you mentioned. Please re-review when available.

Neeharika-Sompalli commented:
> LGTM, aside from the new `equals()`/`hashCode()` methods in classes that don't seem to have any sense of logical identity other than object identity.

Thanks @tinker-michaelj  ! Resolved them as wont fix.

qnswirlds reviewed (approved):
LGTM overall!
